packages/utils/odsp-doclib-utils/src/odspAuth.ts

```diff
@@ -113,9 +113,10 @@ export async function fetchTokens(
 	const refreshToken = tokens.refresh_token;
 
 	if (accessToken === undefined || refreshToken === undefined) {
+		// Look up error codes at https://login.microsoftonline.com/error
 		throwOdspNetworkError(
 			// pre-0.58 error message: unableToGetAccessToken
-			"Unable to get access token",
+			`Unable to get access token. Error code: ${JSON.stringify(tokens.code ?? [])}`,
```

Comment:
This will make these errors non-aggregable in telemetry, and thus hard to estimate impact / prioritize what matters the most.
Please augment error by adding new properties, like it's done in enrichOdspError()

Reply:
sounds good. Btw, do we expect this codepath to be used in prod scenarios? I guess it is used in stress tests. But I would have expected a different auth form for consumer apps.

Reply:
FWIW--to make this helpful, this will also require some refactoring of e2e setup. The logger (initialized in TestObjectProvider) is created after this error is thrown (while constructing the driver)

Reply:
I think that also means this doesn't make it to telemetry currently :)

Reply:
It should not matter if it's used in prod or not - it could change any time, and we do care about error telemetry aggregation for internal test runs as well (i.e. seeing big picture across runs)

Reply:
yes, the prod ask was out of curiosity, not for validity of this change. I believe the actual problem is mitigated at this point, so I'll look instead toward logging errors in test provider initialization and enriching them appropriately.

Reply:
I believe latest iteration addresses this feedback (and also makes sure we're actually logging these errors as well as others that happen during test initialization). resolving.